src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/create-cluster/create-cluster.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { ClusterService } from '~/app/shared/api/cluster.service';
+import { OrchestratorService } from '~/app/shared/api/orchestrator.service';
+import { AppConstants } from '~/app/shared/constants/app.constants';
+import { NotificationType } from '~/app/shared/enum/notification-type.enum';
+import { Permission } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { NotificationService } from '~/app/shared/services/notification.service';
+
+@Component({
+  selector: 'cd-create-cluster',
+  templateUrl: './create-cluster.component.html',
+  styleUrls: ['./create-cluster.component.scss']
+})
+export class CreateClusterComponent implements OnInit {
+  permission: Permission;
+  orchStatus = false;
+  featureAvailable = false;
+  projectConstants: typeof AppConstants;
+
+  constructor(
+    private authStorageService: AuthStorageService,
+    private orchService: OrchestratorService,
+    private router: Router,
+    private clusterService: ClusterService,
+    private notificationService: NotificationService
+  ) {
+    this.permission = this.authStorageService.getPermissions().configOpt;
+  }
+
+  ngOnInit() {
+    this.projectConstants = AppConstants;
+    this.orchService.status().subscribe((status) => {
+      this.orchStatus = status?.available;
+      this.featureAvailable = status.features.add_host?.available;
+      if (!this.orchStatus && !this.featureAvailable) {
+        this.router.navigate(['']);
+      }
+    });
+  }
```

Comment:
In Angular jargon, this is basically a [Router Guard](https://codecraft.tv/courses/angular/routing/router-guards/): a service to decide whether a component is routable based on a condition. We already have ModuleStatusGuardService and we use that for the Orchestrator in several places (you may check `app-routing.module.ts`.

That only checks the status. For checking a specific feature:
* We may extend that guard to support specifying features.
* We may reconsider why we are using a feature (just to distinguish rook from cephadm?). And perhaps there's another more explicit way of doing it.

By reusing existing components, you avoid having to test these behaviors: you can just concentrate on the 'static' component behaviour without having to mess with the preconditions.

Reply:
Used the ModuleStatusGuard. I'l take care of checking a specific feature.

Reply:
Done.